src/examples/graph.js

```diff
@@ -529,31 +555,47 @@ class Graph extends React.Component<IGraphProps, IGraphState> {
       const newIDs = {};
 
       // Every node position is relative to the top and left-most corner
-      const newNodes = copiedNodes.map(copiedNode => {
-        const x = mouseX + (copiedNode.x - cornerX);
-        const y = mouseY + (copiedNode.y - cornerY);
-        const id = `${copiedNode.id}_${Date.now()}`;
-
-        newIDs[copiedNode.id] = id;
-
-        return {
-          ...copiedNode,
-          id,
-          x,
-          y,
-        };
-      });
-
-      const newEdges = copiedEdges.map(copiedEdge => {
-        return {
-          ...copiedEdge,
-          source: newIDs[copiedEdge.source],
-          target: newIDs[copiedEdge.target],
-        };
-      });
+      const newNodes = new Map(
+        Array.from(copiedNodes.values()).map(copiedNode => {
```

Comment:
Personally, the `Array.from(...).map(...)` within `new Map()` feels harder to understand than
```
const newNodes = new Map();

copiedNodes.forEach(copiedNode => {
  ...

  newNodes.set(id, { ... });
});
```

Reply:
Thanks.